subsys/bluetooth/host/settings.c

```diff
@@ -121,6 +121,17 @@ static int set(int argc, char **argv, char *val)
 		return 0;
 	}
 
+	if (!strcmp(argv[0], "name")) {
+		if (strlen(val) >= sizeof(bt_dev.name)) {
+			BT_ERR("Invalid length for device name in storage: name"
+			       " will be truncated");
+		}
+
+		strncpy(bt_dev.name, val, sizeof(bt_dev.name) - 1);
```

Comment:
Shouldn't this be without the `-1`? Otherwise, if you get `strlen(val) == sizeof(bt_dev.name) - 1`  the result of strncpy will not be null terminated (unless `bt_dev.name[sizeof(bt_dev.name - 1)]` has been initialized to zero at some point).

Reply:
Aren't we restoring the value after boot here? I though that would be initialized with 0, note that we still would have to prevent non-null terminated when strlen >= sizeof since that just print a warning and truncates the string, but perhaps in that case we can just set the last byte to '\0' since we know it would be truncated.